src/cbc_sdk/platform/policy_ruleconfigs.py

```diff
@@ -272,3 +299,316 @@ def set_assignment_mode(self, mode):
         if mode not in ("REPORT", "BLOCK"):
             raise ApiError(f"invalid assignment mode: {mode}")
         self.set_parameter("WindowsAssignmentMode", mode)
+
+
+class HostBasedFirewallRuleConfig(PolicyRuleConfig):
+    """Represents a host-based firewall rule configuration in the policy."""
+    swagger_meta_file = "platform/models/policy_ruleconfig.yaml"
+
+    def __init__(self, cb, parent, model_unique_id=None, initial_data=None, force_init=False, full_doc=False):
+        """
+        Initialize the CorePreventionRuleConfig object.
+
+        Args:
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+            parent (Policy): The "parent" policy of this rule configuration.
+            model_unique_id (str): ID of the rule configuration.
+            initial_data (dict): Initial data used to populate the rule configuration.
+            force_init (bool): If True, forces the object to be refreshed after constructing.  Default False.
+            full_doc (bool): If True, object is considered "fully" initialized. Default False.
+        """
+        super(HostBasedFirewallRuleConfig, self).__init__(cb, parent, model_unique_id, initial_data, force_init,
+                                                          full_doc)
+        self._rule_groups = []
+        self._rule_groups_valid = False
+
+    class FirewallRuleGroup(MutableBaseModel):
+        """Represents a group of related firewall rules."""
+        swagger_meta_file = "platform/models/firewall_rule_group.yaml"
+
+        def __init__(self, cb, initial_data):
+            """
+            Initialize the FirewallRuleGroup object.
+
+            Args:
+                cb (BaseAPI): Reference to API object used to communicate with the server.
+                initial_data (dict): Initial data used to populate the firewall rule group.
+            """
+            super(HostBasedFirewallRuleConfig.FirewallRuleGroup, self).__init__(cb, None, initial_data, False, True)
+            self._parent = None
+            self._rules = [HostBasedFirewallRuleConfig.FirewallRule(cb, d)
+                           for d in initial_data.get("rules", [])]
+
+        def _init_parent(self, parent):
```

Comment:
This should be an optional parameter in the constuctor and not implemented as a separate protected function

Reply:
Believe it or not, I originally _had_ this as a constructor parameter, but split this off as the only instance in which the `_parent` references need setting is when we actually load the contents of `_rule_groups`. I guess I should put that back then.

Reply:
This one will be fixed at the next check-in.